modules/EnergyManager/EnergyManager.cpp

```diff
@@ -143,19 +144,37 @@ std::vector<types::energy::EnforcedLimits> EnergyManager::run_optimizer(types::e
 
     for (auto broker : brokers) {
         auto local_market = broker->get_local_market();
+        const auto sold_energy = local_market.get_sold_energy();
 
-        // EVLOG_info << "Sending enfored limits (import) to :" << local_market.energy_flow_request.uuid << " "
-        //            << local_market.get_sold_energy_import()[0].limits_to_root;
-        types::energy::EnforcedLimits l;
-        l.uuid = local_market.energy_flow_request.uuid;
-        l.valid_until =
-            Everest::Date::to_rfc3339(date::utc_clock::now() + std::chrono::seconds(config.update_interval * 10));
+        if (sold_energy.size() > 0) {
 
-        l.limits_root_side = local_market.get_sold_energy()[0].limits_to_root;
+            types::energy::EnforcedLimits l;
+            l.uuid = local_market.energy_flow_request.uuid;
+            l.valid_until =
+                Everest::Date::to_rfc3339(globals.start_time + std::chrono::seconds(config.update_interval * 10));
 
-        l.schedule = local_market.get_sold_energy();
+            l.schedule = sold_energy;
 
-        optimized_values.push_back(l);
+            // select root limit from schedule based on globals.start_time
+            l.limits_root_side = sold_energy[0].limits_to_root;
+
+            for (const auto& s : sold_energy) {
+                const auto schedule_time = Everest::Date::from_rfc3339(s.timestamp);
+                if (globals.start_time < schedule_time) {
+                    // all further schedules will be further into the future
+                    break;
+                } else {
+                    // use this schedule as the starting point
+                    l.limits_root_side = s.limits_to_root;
+                }
+            }
```

Comment:
This looks correct to me. 
If we are in the middle of a schedule, it finds the entry that is closest to now, but still in the past.
If the complete schedule is in the past, it finds the last entry.
If the complete schedule is in the future, it finds the first entry.

Reply:
Tested on a charger and the previously observed delay to changing the charging rate is no longer observed.
Observations prior to the code change
[schedule-timings.pdf](https://github.com/EVerest/everest-core/files/14793267/schedule-timings.pdf)
Observations post to the code change
[schedule-timings-fix.pdf](https://github.com/EVerest/everest-core/files/14793268/schedule-timings-fix.pdf)